base/compiler/optimize.jl

```diff
@@ -393,32 +393,48 @@ function is_known_call(e::Expr, @nospecialize(func), src, spvals::SimpleVector,
     return isa(f, Const) && f.val === func
 end
 
-function renumber_ir_elements!(body::Vector{Any}, changemap::Vector{Int}, preprocess::Bool = true)
-    if preprocess
-        for i = 2:length(changemap)
-            changemap[i] += changemap[i - 1]
+function renumber_ir_elements!(body::Vector{Any}, changemap::Vector{Int})
+    return renumber_ir_elements!(body, changemap, changemap)
+end
+
+function renumber_ir_elements!(body::Vector{Any}, ssachangemap::Vector{Int}, labelchangemap::Vector{Int})
+    for i = 2:length(labelchangemap)
+        labelchangemap[i] += labelchangemap[i - 1]
+    end
+    if ssachangemap !== labelchangemap
+        for i = 2:length(ssachangemap)
+            ssachangemap[i] += ssachangemap[i - 1]
         end
     end
-    changemap[end] != 0 || return
+    (labelchangemap[end] != 0 && ssachangemap[end] != 0) || return
```

Comment:
It seems like this should be `||` instead of `&&`?

Reply:
Hmmm. AFAIK normal use would imply that if either changemap has nonzero entries, then the other has nonzero entries as well. Otherwise, it seems like you've passed invalid input to this function or are using it to apply an invalid pass.  

That's mainly from my own use cases/use cases in Base, though. Maybe there's a use case where you're building up the label changes and SSA changes in separate calls to `renumber_ir_elements`. Not sure what kind of pass that makes sense for, though.